### PR TITLE
Turn strings in headers into bytes. Necessary for Py3 compatibility.

### PR DESCRIPTION
**Pull request by mx**
I was performing a Python3 migration of code that depends on dpkt when I found a problem with bytes(packet) or str(packet). The header packing methods would throw with the following error:
"struct.error: argument for 's' must be a bytes object"

Some other github issues have been created on this same topic for specific packet types; I went through and fixed every example that I could find. This should be a NOP in Python2 (as b'' is an alias for '' there), and fix the above errors in Python3.

### CODE REVIEW

File: dpkt/hsrp.py [hunk 1]
```diff
@@ -39,5 +39,5 @@ class HSRP(dpkt.Packet):
         ('group', 'B', 0),
         ('rsvd', 'B', 0),
         ('auth', '8s', 'cisco'),
```
Comment (obormot) [inline dpkt/hsrp.py]: ```b'cicso'``` ?

**Comment by obormot** [comment]
Good stuff. I'm surprised the unit tests didn't catch any of this. Please add at least one unit test that would fail before the change and work after the change. 

> Some other github issues have been created on this same topic

Please reference these here so we could close them by merging this PR

**Comment by mx** [comment]
> Please reference these here so we could close them by merging this PR
#385 

> Please add at least one unit test that would fail before the change and work after the change.
Done

**Comment by mx** [comment]
Not sure how it thinks coverage *decreased* when I added that test, but I'm gonna guess that's a false positive.

File: dpkt/ethernet.py [hunk 2]
```diff
@@ -717,6 +717,16 @@ def test_eth_802dot1ad_802dot1q_ip():
     assert str(e1) == str(e2)
 
 
+def test_eth_pack():
```
Comment (obormot) [inline dpkt/ethernet.py]: indentation needs to be 4 spaces

Comment (mx) [reply dpkt/ethernet.py]: Done.

**Comment by obormot** [comment]
> Not sure how it thinks coverage _decreased_ when I added that test, but I'm gonna guess that's a false positive.

It decreased because the 2 lines you added in the unit test
```
except:
  raised = True
```
were not executed hence decreased coverage (dumb I know..)

Perhaps restructure the unit test into something like
```
eth = Ethernet(data=b'12345')
assert str(eth)
```

**Comment by mx** [comment]
> > Not sure how it thinks coverage _decreased_ when I added that test, but I'm gonna guess that's a false positive.
> 
> It decreased because the 2 lines you added in the unit test
> 
> ```
> except:
>   raised = True
> ```
> 
> were not executed hence decreased coverage (dumb I know..)
> 
> Perhaps restructure the unit test into something like
> 
> ```
> eth = Ethernet(data=b'12345')
> assert str(eth)
> ```

This feels less clear to the viewer than what's going on there now. I can do something like that if you want, but it feels like making it worse in order to please a metric (coverage).

**Comment by obormot** [comment]
> This feels less clear to the viewer than what's going on there now. I can do something like that if you want, but it feels like making it worse in order to please a metric (coverage).

It's rather to satisfy the metric in order to be able to merge. Can't merge it with decreased coverage

**Comment by mx** [comment]
> > This feels less clear to the viewer than what's going on there now. I can do something like that if you want, but it feels like making it worse in order to please a metric (coverage).
> 
> It's rather to satisfy the metric in order to be able to merge. Can't merge it with decreased coverage

Done, then.

**Comment by mx** [comment]
The remaining coverage failure is because I added the test to main with the other tests.

**Pull request merged**